### DISCUSSION TITLE
Getting elevenlabs extension to work

### DISCUSSION THREAD

**Discussion by St33lMouse** [oobabooga/text-generation-webui#579] [Q&A]
I've installed the eleven labs requirements, but I see no audio interface in chat mode, and no place to enter the api key from eleven labs. Is this working, or am I doing something wrong?

I'm dying to try this out. So close to success!  I don't like the proprietary nature of eleven labs, but their speech is smoother and I think the Silero stuff is just too robotic.

**Comment by St33lMouse** [comment]
Never mind. Loading in the webui wasn't working. Got it working by using command line flags.

**Reply by CMDRBunbun** [reply]
I am having the same issue. Would you mind terribly explaining how you resolved it? I tried adding the command: --extensions [elevenlabs_tts] in the start-webui.bat unfortunately that did not do anything.

**Reply by St33lMouse** [reply]
Ummm don't use the brackets? Also, choose chat mode in the command line. It doesn't work in othere modes and will show no interface.

**Reply by tanfarou** [reply]
Would you please explain where the API key goes? Thanks in advance.

**Reply by Pendragon51** [reply]
could you show an example of the command line, is it --chat?

**Comment by St33lMouse** [comment]
If you successfully load the extension while you are in chat mode you'll see the interface for elevenlabs at the bottom of the chat window. There's a box there, I think.  I'm not at my main computer, can't quite remember.

**Reply by Pendragon51** [reply]
I see where it says API but there is no box to add the api key?

**Comment by BasedAnon** [comment]
> Ummm don't use the brackets? Also, choose chat mode in the command line. It doesn't work in othere modes and will show no interface.
 Brackets where!?!

**Comment by drahmel** [comment]
Got it working.
1. cd to the ~/text-generation-webui/extensions/elevenlabs_tts
2. Install the Eleven Labs library: pip install -r requirements.txt
3. Run the server with the extension activated: python server.py --extensions elevenlabs_tts
4. You now have the Eleven Labs module where you can enter your API key.
![brave_22ifGHc6Ps](https://github.com/oobabooga/text-generation-webui/assets/722938/2a272df3-e25b-4ff9-8e8f-780e4517925a)

**Comment by GiantPopa** [comment]
Could anyone send me a link where i can download the elevenlabs tts extension? I am new to github and cant seem to find it.

**Reply by shaitand** [reply]
Same... where does one get this tts extension?
